.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -46,3 +46,21 @@ If this pull request is unrelated to any issues, please remove
 the following line. -->
 
 Fixes #<Issue Number>
+
+### Checklist for package maintainer(s)
+<!-- This section is to be filled by package maintainer(s) who will
+review this pull request. -->
+
+This checklist is meant to remind the package maintainer(s) who will
+review this pull request of some common things to look for. This list
+is not exhaustive.
+
+- [ ] Do the proposed changes actually accomplish desired goals?
+- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
+- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
+- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
+- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
```

Comment:
~~GitHub also has "Squash and Merge" and "Rebase and Merge" options in the UI now, so I think (hopefully) it will be less often that we ask PR contributors to do that, especially if they aren't already sure how to.  Should we mention that?~~

I see now it's mentioned in the maintainer docs that we should never us it, though I'm not sure why.  It was added quite a few years ago and I don't know if it's still relevant...  Maybe something to look into in another issue.

Reply:
I vaguely remember something about messing up one of the scripts; could be backport or contributor script. Maybe @bsipocz still remembers.

Reply:
Squash-merge doesn't create a merge commit. We rely a lot on merge commits, so it's a no go IMO.
Rebase and merge on the other hand, as I remember at least, may create a merge commit as to resolve the conflicts, so a no go again, and in addition it doesn't test whether the rebased PR still passes CI.

So, afaik, what we need is still missing, e.g. a rebase button, that doesn't create a pull/merge commit, just does a plain and simple rebase and the rerun CI. There are chrome extensions and action bots to do that, but we don' use any of those.